src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -421,6 +421,8 @@ private static void ApplyEndpoints<T>(this IResourceBuilder<T> builder, IResourc
         // Set the environment variable on the resource
         if (env is not null && builder.Resource is IResourceWithEndpoints resourceWithEndpoints and IResourceWithEnvironment)
         {
+            annotation.TargetPortEnvironmentVariable = env;
```

Comment:
It's kind of weird that `WithEndpoint(..., env: "SOME_ENV")` means that `SOME_ENV` gets the "target port" value.

1. Is this assuming too much this deep?
2. Should `env` be renamed?

Reply:
Are you concerned about the name of the variable or the behavior? I tried to remove this feature completely, but it makes executables listening on a port more verbose than it needs to be.

Reply:
> Are you concerned about the name of the variable or the behavior?

A little about the name of the variable. A little about the documentation on what it does. I think I understand the behavior.

Reply:
Updated